guides/0111-container.md

```diff
@@ -0,0 +1,54 @@
+# container
+
+**This block API will likely change in future versions.**
+
+This is a special block which allows to offload computation to a
+[Docker container](https://www.docker.com/resources/what-container). 
+
+Messages are sent and received to/from the container via
+[AMQP](https://www.rabbitmq.com/tutorials/amqp-concepts.html).
+
+The block will manage the creation of the Container in a [Kubernetes](https://kubernetes.io/)
+cluster using the
+[Astarte Kubernetes Operator](https://github.com/astarte-platform/astarte-kubernetes-operator).
+
+
+# Properties
+
+* `image`: Docker container image (required, string)
+* `type`: Container type, either `"producer"`, `"consumer"` or `"producer_consumer"` (required)
+
+## `image`
+
+Docker container image.
+
+## `type`
+
+**This block property is going to be removed in favor of implicit configuration.**
+
+According to container type a container is used as an AMQP producer, cosumer or both.
+
+# Input message
+
+Input messages are consumed only from `consumer` and `producer_consumer` containers.
+
+For those types of containers the input message format depends on their implementation.
```

Comment:
-> The format of the input message depends on the output format of the last block before it